go/cmd/dolt/commands/tblcmds/export.go

```diff
@@ -285,17 +269,83 @@ func NewExportDataMover(ctx context.Context, root *doltdb.RootValue, dEnv *env.D
 		return nil, errhand.BuildDError("Error opening writer for %s.", exOpts.DestName()).AddCause(err).Build()
 	}
 
-	opts := editor.Options{Deaf: dEnv.DbEaFactory()}
-	wr, err := exOpts.dest.NewCreatingWriter(ctx, exOpts, root, srcIsSorted, outSch, statsCB, opts, writer)
-
+	wr, err := exOpts.dest.NewCreatingWriter(ctx, exOpts, root, false, rdSchema, statsCB, editor.Options{Deaf: dEnv.DbEaFactory()}, writer)
 	if err != nil {
-		return nil, errhand.BuildDError("Could not create table writer for %s", exOpts.tableName).AddCause(err).Build()
+		return nil, errhand.BuildDError("Error opening writer for %s.", exOpts.DestName()).AddCause(err).Build()
 	}
 
-	emptyTransColl := pipeline.NewTransformCollection()
+	return wr, nil
+}
+
+func export(ctx context.Context, rd table.TableReadCloser, wr table.TableWriteCloser, vrw types.ValueReadWriter, exOpts *exportOptions) (int64, error) {
+	g, ctx := errgroup.WithContext(ctx)
+
+	parsedRowChan := make(chan sql.Row)
+
+	getNextRow := func(rd table.TableReadCloser) (sql.Row, error) {
```

Comment:
`getNextRow` and `writeRow` make more sense as helper functions than as closures